subsys/net/lib/openthread/platform/radio.c

```diff
@@ -256,6 +260,40 @@ static inline void handle_tx_done(otInstance *aInstance)
 	}
 }
 
+static void openthread_handle_received_frame(otInstance *instance,
+					     struct net_pkt *pkt)
+{
+	otRadioFrame recv_frame;
+
+	recv_frame.mPsdu = net_buf_frag_last(pkt->buffer)->data;
+	/* Length inc. CRC. */
+	recv_frame.mLength = net_buf_frags_len(pkt->buffer);
+	recv_frame.mChannel = platformRadioChannelGet(instance);
+	recv_frame.mInfo.mRxInfo.mLqi = net_pkt_ieee802154_lqi(pkt);
+	recv_frame.mInfo.mRxInfo.mRssi = net_pkt_ieee802154_rssi(pkt);
+
+	if (IS_ENABLED(OPENTHREAD_ENABLE_DIAG) && otPlatDiagModeGet()) {
+		otPlatDiagRadioReceiveDone(instance,
+					   &recv_frame, OT_ERROR_NONE);
+	} else {
+		otPlatRadioReceiveDone(instance,
+				       &recv_frame, OT_ERROR_NONE);
+	}
+
+	net_pkt_unref(pkt);
+}
+
+otError notify_new_rx_frame(struct net_pkt *pkt)
+{
+	otError err = OT_ERROR_NO_BUFS;
+
+	if (k_msgq_put(&rx_pkt_queue, &pkt, K_NO_WAIT) == 0) {
```

Comment:
Perhaps we should block here (`K_FOREVER`) instead of dropping the frame? I think that eventually, the lowest layer (radio driver) should start dropping frames in case upper layers do not catch up processing them. Otherwise, we might end-up dropping frames that are considered "delivered" (acknowledged).

Reply:
In order to achieve that though, a fix in the radio shim would be needed as well (block instead of drop):
https://github.com/zephyrproject-rtos/zephyr/blob/6667dc35e7ea75eb19602ffc86c694ed29587b2c/drivers/ieee802154/ieee802154_nrf5.c#L101